### ISSUE TITLE
Make ShellCliCommandFactory configurable or otherwise permit login shell in Linux

### ISSUE THREAD

**Issue by sarahelsaig** [atata-framework/atata-cli#1]
Currently `ShellCliCommandFactory` is not very flexible on Linux and in certain cases where stuff like `html-validate` won't work:
1. If you set up node via NVM ([recommended](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally) to avoid EACCESS error when trying to install a package "globally" as a regular user) you need to run a login shell, otherwise the NVM setup in the init file (e.g. ~/.bashrc) won't be executed. This is a big deal because if NPM is installed from the package manager then you'd need root access to install using the -g switch.
2. If you don't have bash. This is an edge case but I felt it's also worth mentioning.

It would be nice to introduce a configuration or the ability to replace the default `ShellCliCommandFactory` by replacing the private static field `Atata.Cli.ProgramCli.s_shellCliCommandFactory` with a public property.

For my use case all I need is to replace [one line](https://github.com/atata-framework/atata-cli/blob/main/src/Atata.Cli/ShellCliCommandFactory.cs#L19) like this to include the `-login` switch:
```csharp
                : ("bash", $"-login -c \"{fileNameOrCommand}{EscapeDoubleQuotes(argumentsPart)}\"");
```

**Comment by Piedone** [comment]
@YevgeniyShunevych would it be possible for you to check out this and the corresponding PR?

**Comment by YevgeniyShunevych** [comment]
@Piedone and @DAud-IcI, I got your issue. I will review your PR this week. Maybe will provide an alternative implementation approach.

**Comment by YevgeniyShunevych** [comment]
@Piedone and @DAud-IcI, I published Atata.Cli v1.4 and Atata.HtmlValidation v1.5.0.

With the new functionality you can configure the default CLI command factory the following way:

```cs
ProgramCli.DefaultShellCliCommandFactory = OSDependentShellCliCommandFactory
    .UseCmdForWindows()
    .UseForOtherOS(new BashShellCliCommandFactory("-login"));
```

Another option is to pass a factory through `HtmlValidationOptions`:

```cs
var options = new HtmlValidationOptions
{
    CliCommandFactory = OSDependentShellCliCommandFactory
        .UseCmdForWindows()
        .UseForOtherOS(new BashShellCliCommandFactory("-login"))
};
```

```cs
HtmlValidationOptions.Default = new HtmlValidationOptions
{
    CliCommandFactory = OSDependentShellCliCommandFactory
        .UseCmdForWindows()
        .UseForOtherOS(new BashShellCliCommandFactory("-login"))
};
```

**Comment by YevgeniyShunevych** [comment]
@Piedone and @DAud-IcI, have you tried the above approach? Interesting to know whether it works for you.

**Comment by Piedone** [comment]
Sorry to drop this, Yevgeniy. Dávid will come back to you soon.

**Comment by DAud-IcI** [comment]
Sorry for the delay. I finally had the opportunity to try it and works like charm! First I removed my clunky local workaround so the tests fail again with "html-validate not found". Then I upgraded to Atata 2.0.0 and added the `ProgramCli.DefaultShellCliCommandFactory` configuration you suggested to the setup method and now the tests pass again.

I have closed my PR as it's not needed any more. Thanks again.
